api/turing/config/config.go

```diff
@@ -100,7 +107,58 @@ type DeploymentConfig struct {
 
 // EnsemblingJobConfig captures the config related to the ensembling batch jobs.
 type EnsemblingJobConfig struct {
-	DefaultEnvironment string `validate:"required"`
+	BatchSize          int          `validate:"required"`
+	MaxRetryCount      int          `validate:"required"`
+	DefaultEnvironment string       `validate:"required"`
+	KanikoConfig       KanikoConfig `validate:"required"`
+	ImageConfig        ImageConfig  `validate:"required"`
+	InjectGojekConfig  bool
+}
+
+// ImageConfig provides the configuration used for the OCI image building.
+type ImageConfig struct {
+	LocalRegistry        string
+	Registry             string `validate:"required"`
+	BaseImageRef         string `validate:"required"`
+	BuildNamespace       string `validate:"required"`
+	BuildContextURI      string `validate:"required"`
+	DockerfileFilePath   string `validate:"required"`
+	BuildTimeoutDuration time.Duration
+}
+
+// ResourceRequestsLimits contains the Kubernetes resource request and limits for kaniko
+type ResourceRequestsLimits struct {
+	Requests Resource `validate:"required"`
+	Limits   Resource `validate:"required"`
+}
+
+// Resource contains the Kubernetes resource request and limits
+type Resource struct {
```

Comment:
Nitpick: This can go above the `ResourceRequestsLimits` struct definition which uses it.

Reply:
ok